### PR TITLE
Add Wider Face format support

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Summary

Added support for Wider Face format

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/plugins/wider_format.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    IMAGE_EXT = '.jpg'
+    ANNOTATIONS_DIR = 'wider_face_split'
+    IMAGES_DIR = 'images'
+    SUBSET_DIR = 'WIDER_'
+    BBOX_ATTRIBUTES = ['blur', 'expression', 'illumination',
+        'occlusion', 'pose', 'invalid']
+
+class WiderExtractor(SourceExtractor):
+    def __init__(self, path):
+        assert osp.isfile(path), path
```
Comment (zhiltsov-max) [inline datumaro/plugins/wider_format.py:25]: It is better to provide some meaningful error message.

File: datumaro/plugins/wider_format.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import os
+import os.path as osp
+import re
+
+from datumaro.components.converter import Converter
+from datumaro.components.extractor import (AnnotationType, Bbox, DatasetItem,
+    Importer, SourceExtractor)
+
+
+class WiderPath:
```
Comment (zhiltsov-max) [inline datumaro/plugins/wider_format.py:15]: I think `Wider` is not clear enough and suggest using `Widerface` instead.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Dataset Management Framework (Datumaro)
+d# Dataset Management Framework (Datumaro)
```
Comment (zhiltsov-max) [inline README.md:1]: Extra symbols.

File: datumaro/plugins/wider_format.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            if WiderPath.IMAGE_EXT in line]
+
+        for image_id in image_ids:
+            image = lines[image_id]
+            image_path = osp.join(self._dataset_dir, WiderPath.SUBSET_DIR
+                + self._subset, WiderPath.IMAGES_DIR, image[:-1])
+            item_id = image[:-(len(WiderPath.IMAGE_EXT) + 1)]
+
+            bbox_count = lines[image_id + 1]
+            bbox_lines = lines[image_id + 2:image_id + 2 + int(bbox_count)]
```
Comment (zhiltsov-max) [inline datumaro/plugins/wider_format.py:52]: Please, maintain the formatting:
- spaces in slices
- ascending direction in comparisons (<, <=)

File: datumaro/plugins/wider_format.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        os.makedirs(save_dir, exist_ok=True)
+
+        for subset_name, subset in self._extractor.subsets().items():
+            subset_dir = osp.join(save_dir, WiderPath.SUBSET_DIR + subset_name)
+
+            wider_annotation = ''
+            for item in subset:
+                wider_annotation += '%s\n' % (item.id + WiderPath.IMAGE_EXT)
+                if self._save_images:
+                    self._save_image(item, osp.join(save_dir, subset_dir,
```
Comment (zhiltsov-max) [inline datumaro/plugins/wider_format.py:94]: Please, add `.image` existence check with `.has_image`.

Comment (yasakova-anastasia) [reply datumaro/plugins/wider_format.py:94]: Why add this check if it already exists in `_save_image`?

Comment (zhiltsov-max) [reply datumaro/plugins/wider_format.py:94]: `_save_image` checks for image data, but it expects an image. It is possible that input data does not have image info at all, just annotations. A good question, though. Maybe, this check is worth adding there as well.

**Review by zhiltsov-max — APPROVED** [review]

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**